### PR TITLE
libvirt: 8.5.0 -> 8.6.0, bugfixes

### LINKED ISSUE #165897: libvirt: session-mode libvirt can't always find /run/libvirt/nix-helpers/qemu-bridge-helper
### Describe the bug
I sometimes create a qemu VM in libvirt in "session mode", meaning that the libvirt daemon is running as my non-root user, and the system-wide libvirtd does not need to be running ([docs](https://libvirt.org/daemons.html#operating-modes)). This VM has a network bridge, which libvirt apparently creates by calling `/run/wrappers/bin/qemu-bridge-helper`, which in turn uses `/run/libvirt/nix-helpers/qemu-bridge-helper`. 

The problem is that `/run/libvirt` doesn't get fully populated until the system libvirtd gets started. Previously, in nixpkgs 7f9b6e2ba (from February), this wasn't noticeable because libvirtd.service was WantedBy libvirt-guests.service (which is WantedBy multi-user.target), so libvirtd would always be started on boot to populate that directory. Now, with 1d08ea2b, this is no longer the case, and libvirtd.service does not get started (presumably, if libvirt-guests.service actually needed to start a VM, then libvirtd.service would be socket-activated, but for me this is not the case). As a result, `/run/libvirt` doesn't get populated and my session-mode libvirt can't find `/run/libvirt/nix-helpers/qemu-bridge-helper`. 

### Steps To Reproduce
1. In python:
   ```python
   conn = libvirt.open('qemu:///session')
   dom = conn.createXML(Path('path/to/vm.xml').read_text())
   ```
   The VM's XML file defines a network bridge, which is why `qemu-bridge-helper` gets involved.
3. Get the following output:
   ```
   libvirt: QEMU Driver error : internal error: /run/wrappers/bin/qemu-bridge-helper --use-vnet --br=virbr0 --fd=29: failed to communicate with bridge helper: Transport endpoint is not connected
   stderr=/run/wrappers/bin/qemu-bridge-helper: cannot run `/run/libvirt/nix-helpers/qemu-bridge-helper': No such file or directory
   ...
   libvirt.libvirtError: internal error: /run/wrappers/bin/qemu-bridge-helper --use-vnet --br=virbr0 --fd=29: failed to communicate with bridge helper: Transport endpoint is not connected
   stderr=/run/wrappers/bin/qemu-bridge-helper: cannot run `/run/libvirt/nix-helpers/qemu-bridge-helper': No such file or directory
   ```

### Expected behavior
The libvirt user session should be usable without first starting the system-wide libvirtd.

### Additional context
If you start (and optionally stop) the system libvirtd.service before following the steps to reproduce, everything works fine, because the `/run/libvirt` directory gets populated. 

Probably, `/run/libvirt/nix-helpers/qemu-bridge-helper` should be put somewhere else, where session-mode daemons can use it without needing to have previously started system-mode libvirtd. (The same may be true of other stuff in `/run/libvirt` too.)

### Notify maintainers
libvirt maintainers: @lovesegfault, @fpletz, @globin 

Okay, but I'm using flakes and that channel looks old. See relevant commit hashes above.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.0, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0pre20220322_d5d4d98`
 - channels(root): `"nixos-21.11pre333218.263ef4cc414, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### LINKED ISSUE #182835: libvirtd: "Cannot find 'parted' in path" when creating volume
### Describe the bug
When attempting to create a volume on a physical node pool parted is not found.

```console
$ sudo virsh vol-create-as nvme0n1 nixos-unstable 1863GB
error: Failed to create vol nixos-unstable
error: Cannot find 'parted' in path: No such file or directory
```

### Steps To Reproduce
Create a physical disk node pool.
Attempt to create a volume using that node pool.
```
sudo virsh vol-create-as nvme0n1 nixos-unstable 1863GB
```
In this case I have created a node pool called `nvme0n1` and I am attempting to create a volume called `nixos-unstable`

### Expected behavior
Volume is created normally.

### Additional context
I have tried to add parted manually using and overriding the package in the libvirtd module:
```nix
  libvirt = pkgs.libvirt.overrideAttrs (old: {
    buildInputs = old.buildInputs ++ [ pkgs.parted ];
  });
```
but this did not fix the issue.

### Notify maintainers
@fpletz 
@globin
@lovesegfault

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m
∙ "
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.53, NixOS, 22.11 (Raccoon)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(collin): `""`
 - channels(root): `"nixos-21.11.335130.386234e2a61"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by fpletz**
###### Description of changes

- Fixes #182835
- Fixes #165897
- Closes #180647

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libvirt/default.nix [hunk 1]
```diff
@@ -133,6 +133,8 @@ stdenv.mkDerivation rec {
     sed -i '/virnetsockettest/d' tests/meson.build
     # delete only the first occurrence of this
     sed -i '0,/qemuxml2argvtest/{/qemuxml2argvtest/d;}' tests/meson.build
+  '' + optionalString isLinux ''
+    sed -i 's,define PARTED "parted",define PARTED "${parted}/bin/parted",' src/storage/storage_backend_disk.c
```
[the remark this answers is not in the document]
Comment (fpletz) [reply pkgs/development/libraries/libvirt/default.nix:137]: Thanks, I'll check and add this. Thinking about adding a NixOS test...

File: nixos/modules/virtualisation/libvirtd.nix [hunk 2]
```diff
@@ -282,7 +282,7 @@ in
       setuid = true;
       owner = "root";
       group = "root";
-      source = "/run/${dirName}/nix-helpers/qemu-bridge-helper";
+      source = "${cfg.qemu.package}/libexec/qemu-bridge-helper";
```
[the remark this answers is not in the document]
Comment (fpletz) [reply nixos/modules/virtualisation/libvirtd.nix:285]: I was wondering the same thing when I added this. The package itself seems to use the SUID wrapper anyway. Will try to remove it.

[the remark this answers is not in the document]
Comment (fpletz) [reply nixos/modules/virtualisation/libvirtd.nix:285]: Added a test. Not needed anymore. :smile_cat:

File: pkgs/development/libraries/libvirt/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (fpletz) [reply pkgs/development/libraries/libvirt/default.nix]: Fixed. Thanks!

**Comment by fpletz** [comment]
@GrahamcOfBorg test libvirtd

**Review by Ma27 — APPROVED** [review]

**Comment by fpletz** [comment]
Increased the memory so the test won't timeout on `aarch64-linux`. Also, shutdown doesn't seem to work on `aarch64-linux`, so we just destroy the domain.

**Pull request merged**